collect_app/src/main/java/org/odk/collect/android/tasks/sms/SmsService.java

```diff
@@ -400,4 +410,59 @@ public static String getDisplaySubtext(int resultCode, Date date, SmsProgress pr
 
         return "";
     }
+
+    /**
+     * Automatically submits finalized instances that have sms tags.
+     * Before submission there are checks to see if
+     *
+     * 1) The transport is set to SMS
+     * 2) SMS auto send is enabled.
+     * 3) The form's auto send option isn't overriding the app settings.
+     * 4) SMS tags are included in the form submission.
+     */
+    public void autoSendForms() {
+        String transport = (String) GeneralSharedPreferences.getInstance().get(KEY_SUBMISSION_TRANSPORT_TYPE);
+        boolean smsTransportEnabled = transport.equalsIgnoreCase(context.getString(R.string.transport_type_value_sms));
+
+        if (!smsTransportEnabled) {
+            return;
+        }
+
+        boolean autoSend = false;
+        Set<String> multiAutoSend = (Set<String>) GeneralSharedPreferences.getInstance().get(PreferenceKeys.KEY_MULTI_AUTOSEND);
+
+        for (String option : multiAutoSend) {
```

Comment:
Same as above, `contains` is a better choice. `Set`s do retrievals in constant time so it's very odd to see iteration over them!

Reply:
I didn't like the iteration approach you know but I just didn't remember `contains` method. Thanks!